renderer/tasks/rename.js

```diff
@@ -19,7 +19,7 @@ module.exports = function(project) {
 
         // iterate over each file and create rename(move) callback
         for (let asset of project.assets) {
-            let src = path.join( project.workpath, path.basename(url.parse(asset.src).pathname));
+            let src = path.join( project.workpath, asset.src.substring( asset.src.lastIndexOf('/') + 1 ));
```

Comment:
I'm glad you asked, because while writing up an explanation I realized my original reasoning for this change was incorrect. I made this change to solve an issue I was having with spaces in file names, but now that I think through it again I don't think this was the best way to fix it.

If you look at my first commit, I was calling `downloadFromS3(asset.bucket, asset.key, project.workpath, asset.name)` - this meant if I had a file like `my video.mp4` it would get downloaded and saved as `my video.mp4`, but then `path.basename(url.parse(asset.src).pathname)` in `rename.js` would look for `my%20video.mp4` (because `url.parse` encodes spaces as `%20`) and the project would fail at that stage.

I changed both pieces to use `asset.src.substring( asset.src.lastIndexOf('/') + 1 )` to grab the file name out of the URL without using `url.parse` and that fixed my issue. However, now that I think about it I probably should have just changed my call to `downloadFromS3` to use `url.parse` as well, like so: `downloadFromS3(asset.bucket, asset.key, project.workpath, path.basename(url.parse(asset.src).pathname))` then there would be no change required to `rename.js`

Does that make sense? I'm going to push another commit with this change so you can see what I mean. Changing this should fix the failing tests as well (my mistake for not running the tests before submitting the PR).